tst/testinstall/MatrixObj/testmatobj_new.g

```diff
@@ -0,0 +1,299 @@
+    
+    if nrCols <> fail then
+        if nrCols <> ex.nrCols then
+            MatObjTest_HandleErrorWrongResult("NrCols", [ex.matObj], opt.breakOnError, ex, errors);
+        fi;
+    fi; 
+end;
+
+MatObjTest_TestMatElm := function(ex, opt, errors)
+    local col, row, elm;
+
+    col := ex.nrCols;
+    row := ex.nrRows;
+    elm := MatObjTest_CallFunc(MatElm, [ex.matObj, row, col], opt.breakOnError, errors, ex);
+
+    if elm <> fail then
+        if elm <> ex.sourceOfTruth[row,col] then
+            MatObjTest_HandleErrorWrongResult("MatElm", [ex.matObj, row, col], opt.breakOnError, ex, errors);
+        fi;
+    fi;
+end;
+
+MatObjTest_TestSetMatElm := function(ex, opt, errors)
+    local col, row, elm;
+
+    col := ex.nrCols;
+    row := ex.nrRows;
+    elm := Zero(ex.baseDomain);
+    ex.sourceOfTruth[row, col] := elm;
+    ex.mat[row, col] := elm;
+    MatObjTest_CallFunc(SetMatElm, [ex.matObj, row, col, elm], opt.breakOnError, errors, ex);
+
+    if ex.matObj[row, col] <> ex.sourceOfTruth[row, col] then
+        MatObjTest_HandleErrorWrongResult("SetMatElm", [ex.matObj, row, col, elm], opt.breakOnError, ex, errors);
+    fi;
+end;
+
+MatObjTest_TestConstructingFilter := function(ex, opt, errors)
+    local filter;
+    #Error("before CallFunc");
+    filter := MatObjTest_CallFunc(ConstructingFilter, [ex.matObj], opt.breakOnError, errors, ex);
+    #Error("made it past CallFunc");
+    if filter <> fail then
+        if filter <> ex.filter then
+            MatObjTest_HandleErrorWrongResult("ConstructingFilter", [ex.matObj], opt.breakOnError, ex, errors);
+        fi;
+    fi;
+end;
+
+
+TestMatrixObj := function(filter, opt)
+    local errors, examples, ex;
+
+    errors := [];
+
+    if not IsBound(opt.randomSource) then
+        opt.randomSource := GlobalMersenneTwister;
+    fi;
+
+    if not IsBound(opt.state) then 
+        opt.state := State(GlobalMersenneTwister);
+    fi;
+
+    if not IsBound(opt.forbidInv) then
+        opt.forbidInv := false;
+    fi;
+
+    if not IsBound(opt.breakOnError) then
+        opt.breakOnError := false;
+    fi;
+
+    examples := MatObjTest_GenerateExample(filter, opt);
+
+    for ex in examples do
+        MatObjTest_TestBaseDomain(ex, opt, errors);
+    od;
+
+    for ex in examples do
+        MatObjTest_TestNrRows(ex, opt, errors);
+    od; 
+
+    for ex in examples do
+        MatObjTest_TestNrCols(ex, opt, errors);
+    od; 
+
+    for ex in examples do
+        MatObjTest_TestMatElm(ex, opt, errors);
+    od; 
+
+    for ex in examples do
+        MatObjTest_TestSetMatElm(ex, opt, errors);
+    od; 
+
+    for ex in examples do
+        MatObjTest_TestConstructingFilter(ex, opt, errors);
+    od;
+
+    #TODO other tests
+
+    if not opt.forbidInv then
+        opt.invertible := true;
+        examples := MatObjTest_GenerateExample(filter, opt);
+
+        for ex in examples do
+            #TestInverse(ex);
```

Comment:
We may also wish to do fewer tests (and with smaller dimensions) for a slower operation like `Inverse` or `MinimalPolynomial` compared to somethign very fast with `NrRows`

Reply:
There now is a specific option to specify a different set of dimensions and domains to use for expensive tests. There is even an option to provide a specific list of examples to use in these cases.